src/common/transformations/src/transformations/common_optimizations/fq_reshape_fusion.cpp

```diff
@@ -62,21 +66,41 @@ ov::pass::FakeQuantizeReshapeFusion::FakeQuantizeReshapeFusion() {
                                });
                 const auto& new_limit_size = shape_size(new_limit_shape);
                 if (new_limit_size == limit_size) {  // we tracked future channel placement
-                    if (new_limit_shape == limit_input.get_shape())
+                    if (new_limit_shape == limit_input.get_shape()) {
                         renewed_inputs.push_back(limit_input);
-                    else
-                        renewed_inputs.push_back(reshape_node->clone_with_new_inputs(
+                    } else {
+                        auto reshaped_input = reshape_node->clone_with_new_inputs(
                             {limit_input,
-                             ov::op::v0::Constant::create(element::i64, {new_limit_shape.size()}, new_limit_shape)}));
+                             ov::op::v0::Constant::create(element::i64, {new_limit_shape.size()}, new_limit_shape)});
+                        if (auto constant = ov::util::get_constant_from_source(reshaped_input)) {
+                            reshaped_input = constant;
+                        }
+                        renewed_inputs.push_back(reshaped_input);
+                    }
                     continue;
                 }
             }
             // resulting FQ will become or already is more than per-tensor / per-channel
             return false;
         }
+
+        auto reshaped_input =
+            reshape_node->clone_with_new_inputs({pattern_map.at(data_p), reshape_node->input_value(1)});
+        if (auto constant = ov::util::get_constant_from_source(reshaped_input)) {
+            reshaped_input = constant;
+        }
+        if (pattern_map.count(convert_p)) {
+            const auto& convert_node = pattern_map.at(convert_p).get_node_shared_ptr();
+            convert_node->input(0).replace_source_output(reshaped_input);
+            convert_node->validate_and_infer_types();
```

Comment:
Yes, you ARE right. Since we are in a single Graph Rewrite -- it is safer to keep the graph in validated state, so that all the matchers could assume rank and shape is accurate